src/vmm/src/persist.rs

```diff
@@ -67,6 +71,23 @@ pub struct MicrovmState {
     pub device_states: DeviceStates,
 }
 
+/// This describes the mapping between Firecracker base virtual address and offset in the
+/// buffer or file backend for a guest memory region. It is used to tell an external
+/// process/thread where to populate the guest memory data for this range.
+///
+/// E.g. Guest memory contents for a region of `size` bytes can be found in the backend
+/// at `offset` bytes from the beginning, and should be copied/populated into `base_host_address`.
+#[derive(Clone, Debug, Serialize)]
+pub struct RegionBackendMapping {
```

Comment:
`RegionBackendMapping` seems too generic to me because the backend is actually very specific here. Will every backend need this info ? For example the file backend doesn't need this structure.

Reply:
Sounds good to me